src/Raven.Client/Documents/Queries/QueryResult.cs

```diff
@@ -21,10 +21,15 @@ public abstract class QueryResult<TResult, TIncludes> : QueryResultBase<TResult,
         /// </summary>
         public int TotalResults { get; set; }
 
-       /// <summary>
-       /// The total results for the query, taking into account the 
-       /// offset / limit clauses for this query
-       /// </summary>
+        /// <summary>
+        /// Gets or sets the total results as int64 type for this query
+        /// </summary>
+        public long TotalResults64 { get; set; }
```

Comment:
Better to call it LongTotalResults to match .NET ? https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.longcount?view=net-5.0

Reply:
Done